lib/push.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var request = require('request')
+
+var ERR_NO_PUSH_CALLBACK = 'No Push Callback'
+
+var LOG_UPDATE_INCREMENT_SEND = 'push.send'
+var LOG_UPDATE_INCREMENT_SUCCESS = 'push.success'
+var LOG_UPDATE_INCREMENT_FAIL = 'push.failed'
+var LOG_UPDATE_INCREMENT_NO_CALLBACK = 'push.no_push_callback'
+var LOG_OP_NOTIFY_UPDATE = 'push.notifyUpdate'
+
+module.exports = function (log, db) {
+  return {
+    /**
+     *  Notifies all devices that there was an update to the account
+     *
+     * @param uid
+     * @promise
+     */
+    notifyUpdate: function notifyUpdate(uid) {
+      return db.devices(uid).then(
+        function (devices) {
+          log.trace({op: LOG_OP_NOTIFY_UPDATE, devices: devices})
+
+          devices.forEach(function (device) {
+            if (device.pushCallback) {
+              log.trace({op: LOG_OP_NOTIFY_UPDATE, pushCallback: device.pushCallback})
+
+              // send the push notification
+              log.increment(LOG_UPDATE_INCREMENT_SEND)
+              request.post(device.pushCallback, function (err){
+                if (err) {
+                  log.error({op: LOG_OP_NOTIFY_UPDATE, err: err})
+                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                } else {
+                  log.increment(LOG_UPDATE_INCREMENT_SUCCESS)
+                }
+              })
+            } else {
+              // keep track if there are any devices with no push urls.
+              log.error({op: LOG_OP_NOTIFY_UPDATE, err: ERR_NO_PUSH_CALLBACK})
```

Comment:
I wanted to make sure this logs an error because it is important to not have these errors for the device flow. Mentioning to @jrgm right now: if there are devices with no push callback urls then we will get errors from the auth server.